http-server-netty/src/main/java/io/micronaut/http/server/netty/RoutingInBoundHandler.java

```diff
@@ -216,42 +183,51 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
             return;
         }
 
-        NettyHttpRequest<?> nettyHttpRequest = NettyHttpRequest.remove(ctx);
-        if (nettyHttpRequest == null) {
-            if (cause instanceof SSLException || cause.getCause() instanceof SSLException) {
-                if (LOG.isDebugEnabled()) {
-                    LOG.debug("Micronaut Server Error - No request state present. Cause: " + cause.getMessage(), cause);
-                }
-            } else {
-                if (LOG.isErrorEnabled()) {
-                    LOG.error("Micronaut Server Error - No request state present. Cause: " + cause.getMessage(), cause);
-                }
+        if (cause instanceof SSLException || cause.getCause() instanceof SSLException) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Micronaut Server Error - No request state present. Cause: " + cause.getMessage(), cause);
+            }
+        } else {
+            if (LOG.isErrorEnabled()) {
+                LOG.error("Micronaut Server Error - No request state present. Cause: " + cause.getMessage(), cause);
             }
-
-            ctx.writeAndFlush(new DefaultFullHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.INTERNAL_SERVER_ERROR));
-            return;
         }
-        new NettyRequestLifecycle(this, ctx, nettyHttpRequest).handleException(cause);
     }
 
     @Override
-    protected void channelRead0(ChannelHandlerContext ctx, io.micronaut.http.HttpRequest<?> httpRequest) {
-        new NettyRequestLifecycle(this, ctx, (NettyHttpRequest<?>) httpRequest).handleNormal();
+    public void accept(ChannelHandlerContext ctx, io.netty.handler.codec.http.HttpRequest request, PipeliningServerHandler.OutboundAccess outboundAccess) {
+        NettyHttpRequest<Object> mnRequest;
+        try {
+            mnRequest = new NettyHttpRequest<>(request, ctx, conversionService, serverConfiguration);
+        } catch (IllegalArgumentException e) {
+            NettyHttpRequest<Object> errorRequest = new NettyHttpRequest<>(
```

Comment:
What is this scenario?

Reply:
parse error of the URI

Reply:
i will add a comment here and in other places